RecoLocalFastTime/FTLCommonAlgos/interface/FTLRecHitAlgoBase.h

```diff
@@ -0,0 +1,50 @@
+#ifndef RecoLocalFastTime_FTLCommonAlgos_FTLRecHitAlgoBase_HH
+#define RecoLocalFastTime_FTLCommonAlgos_FTLRecHitAlgoBase_HH
+
+/** \class FTLRecHitAlgoBase
+  *  Template algorithm to make rechits from uncalibrated rechits
+  *
+  *  \author Lindsey Gray
+  */
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+#include "DataFormats/FTLRecHit/interface/FTLRecHit.h"
+#include "DataFormats/FTLRecHit/interface/FTLUncalibratedRecHit.h"
+
+namespace edm {
+  class Event;
+  class EventSetup;
+}
+
+class FTLRecHitAlgoBase {
+ public:
+
+  /// Constructor
+  FTLRecHitAlgoBase(const edm::ParameterSet& conf,
+                    edm::ConsumesCollector& sumes):
+    name_( conf.getParameter<std::string>("algoName") ){ };
+
+  /// Destructor
+  virtual ~FTLRecHitAlgoBase() { };
+
+  /// get event and eventsetup information
+  virtual void getEvent(const edm::Event&) = 0;
```

Comment:
Hi @lgray - i found these names confusing - these are methods that set things using the event(setup). You might consider their names in a future pr.